ending_location_visitor.py

```diff
@@ -1,32 +1,259 @@
+        # TODO: attach more nodes with their transform functions here.
+        self.ending_transformer.register_transform(astroid.Arguments, set_general)
+
+        self.ending_transformer.register_transform(astroid.Const, set_general)
+        self.ending_transformer.register_transform(astroid.BinOp, set_general)
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+        
+        
+
+    def setUp(self):
+        """Method called to prepare the test fixture. This is called immediately
+        before calling the test method; other than AssertionError or SkipTest,
+        any exception raised by this method will be considered an error rather
+        than a test failure. The default implementation does nothing."""
+        pass
+
+    def tearDown(self):
+        """Method called immediately after the test method has been called and
+        the result recorded. This is called even if the test method raised an
+        exception, so the implementation in subclasses may need to be
+        particularly careful about checking internal state. Any exception,
+        other than AssertionError or SkipTest, raised by this method will be
+        considered an additional error rather than a test failure (thus
+        increasing the total number of reported errors). This method will only
+        be called if the setUp() succeeds, regardless of the outcome of the
+        test method. The default implementation does nothing."""
+        self.ending_visitor.reset()
+
+    def _get_file_as_module(self, file_location):
+        """Given a filepath (file_location), parse with astroid, and return 
+        the module.
+        """
+        with open(file_location) as f:
+            content = f.read()
+        return self._get_string_as_module(content)
+
+    def _get_string_as_module(self, string):
+        """Parse the string with astroid, and return the module.
+        """
+        return astroid.parse(string)
+
+    def _are_equal(self, expected, props_assigned):
+        """Check if two lists are equal. Not more items, and not less items.
+        Note we cannot use a set because this would hide the inequality from
+        repeated items.
+        TODO: check for equality without respect to order of items in top-level.
+        For example, the following lists are 'equal' for our purposes.
+        [[1, 1, 0, 6], [2, 2, 4, 6]]
+        [[2, 2, 4, 6], [1, 1, 0, 6]]
+        """
+        return expected == props_assigned
+
+    def assertSameness(self, expected, props):
+        """If sequences are not equal, it is convenient to see each side.
+        """
+        try:
+            self.assertTrue(self._are_equal(expected, props))
+        except AssertionError as e:
+            logging.error('Compare:\n{}\nprops: {}'.format('-'*70, props))
+            logging.error('expected: {}'.format(expected))
+            logging.error('[lineno, end_lineno, col_offset, end_col_offset]')
+            raise e  # otherwise, test will always 'pass'
+
+    def _all_props_set(self):
```

Comment:
Please remove this for now. Once more types are supported, we can add it back in.